thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/algorithm/BaselineRuleFilterWrapper.java

```diff
@@ -0,0 +1,98 @@
+ */
+public class BaselineRuleFilterWrapper extends RuleBasedFilterWrapper {
+  private static final String PROP_WEEKS = "weeks";
+  private static final int PROP_WEEKS_DEFAULT = 1;
+
+  private static final String PROP_CHANGE = "change";
+  private static final double PROP_CHANGE_DEFAULT = Double.NaN;
+
+  private static final String PROP_DIFFERENCE = "difference";
+  private static final double PROP_DIFFERENCE_DEFAULT = Double.NaN;
+
+  private static final String PROP_TIMEZONE = "timezone";
+  private static final String PROP_TIMEZONE_DEFAULT = "UTC";
+
+  private static final String PROP_SITEWIDE_METRIC = "siteWideMetricUrn";
+  private static final String PROP_SITEWIDE_THRESHOLD = "siteWideImpactThreshold";
+  private static final double PROP_SITEWIDE_THRESHOLD_DEFAULT = Double.NaN;
+
+  private Baseline baseline;
+  private double change;
+  private double difference;
+  private double siteWideImpactThreshold;
+  private String siteWideMetricUrn;
+
+  public BaselineRuleFilterWrapper(DataProvider provider, DetectionConfigDTO config, long startTime, long endTime) {
+    super(provider, config, startTime, endTime);
+    int weeks = MapUtils.getIntValue(config.getProperties(), PROP_WEEKS, PROP_WEEKS_DEFAULT);
+    DateTimeZone timezone =
+        DateTimeZone.forID(MapUtils.getString(this.config.getProperties(), PROP_TIMEZONE, PROP_TIMEZONE_DEFAULT));
+    this.baseline = BaselineAggregate.fromWeekOverWeek(BaselineAggregateType.MEDIAN, weeks, 1, timezone);
+    // percentage change
+    this.change = MapUtils.getDoubleValue(config.getProperties(), PROP_CHANGE, PROP_CHANGE_DEFAULT);
+    // absolute change
+    this.difference = MapUtils.getDoubleValue(config.getProperties(), PROP_DIFFERENCE, PROP_DIFFERENCE_DEFAULT);
+    // site wide impact
+    this.siteWideImpactThreshold = MapUtils.getDoubleValue(config.getProperties(), PROP_SITEWIDE_THRESHOLD, PROP_SITEWIDE_THRESHOLD_DEFAULT);
+    this.siteWideMetricUrn = MapUtils.getString(config.getProperties(), PROP_SITEWIDE_METRIC);
+  }
+
+  @Override
+  boolean isQualified(MergedAnomalyResultDTO anomaly) {
+    MetricEntity me = MetricEntity.fromURN(anomaly.getMetricUrn());
+    MetricSlice currentSlice =
+        MetricSlice.from(me.getId(), anomaly.getStartTime(), anomaly.getEndTime(), me.getFilters());
+    MetricSlice baselineSlice = this.baseline.scatter(currentSlice).get(0);
+
+    Map<MetricSlice, DataFrame> aggregates = this.provider.fetchAggregates(Arrays.asList(currentSlice, baselineSlice), Collections.<String>emptyList());
+    double currentValue = getValueFromAggregates(currentSlice, aggregates);
+    double baselineValue = getValueFromAggregates(baselineSlice, aggregates);
+    if (!Double.isNaN(this.difference) && Math.abs(currentValue - baselineValue) < this.difference) {
+      return false;
+    }
+    if (!Double.isNaN(this.change) && baselineValue != 0 && Math.abs(currentValue / baselineValue - 1) < this.change) {
```

Comment:
... and here

Reply:
Yeah. If the dividend is zero, the division will be short-circuited. so this anomaly will not be filtered.